providers/FeedProvider.tsx

```diff
@@ -47,6 +54,7 @@ const FeedProvider = ({ children }: { children: ReactNode }) => {
   }, [userLoading, user]);
 
   const updateFilter = (change: FeedFilter) => {
+    setIsChannelChanged(true);
```

Comment:
hmmm... this feels like a code smell
- we shouldn't need an extra state variable for `isChannelChanged`

gonna look more into your code to understand why you added a new state variable.